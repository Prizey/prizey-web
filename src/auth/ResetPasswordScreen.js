import React from 'react'
import { Redirect } from '@reach/router'
import { ResetPassword } from 'croods-auth'
import { NewPassword } from 'seasoned-auth-forms-web'

import Layout from 'components/Layout/Layout'

export const appendToken = (create, location) => values => {
  const urlParams = new URLSearchParams(location.search)

  return create({
    ...values,
    resetPasswordToken: urlParams.get('reset_password_token'),
  })
}

export const Form = ({ create, creating, error, location }) => (
  <Layout>
    <NewPassword
      onSubmit={appendToken(create, location)}
      submitting={creating}
      submitError={error}
    />
  </Layout>
)

export default props => (
  <ResetPassword
    {...props}
    setCurrentUser={() => {}}
    render={renderProps => <Form {...renderProps} />}
    renderCreated={() => <Redirect to="/" noThrow />}
  />
)
