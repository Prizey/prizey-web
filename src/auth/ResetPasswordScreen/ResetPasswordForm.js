import React from 'react'
import { NewPassword } from 'seasoned-auth-forms-web'

import SubmitButton from 'design/SubmitButton/SubmitButton'

export default props => (
  <NewPassword
    {...props}
    renderButton={buttonProps => (
      <SubmitButton {...buttonProps} label="SET NEW PASSWORD" />
    )}
  />
)
