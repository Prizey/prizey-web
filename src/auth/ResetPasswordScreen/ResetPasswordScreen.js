import React from 'react'
import { Redirect } from '@reach/router'
import { ResetPassword } from 'croods-auth'
import ResetPasswordForm from './ResetPasswordForm'

import createAuthForm from '../AuthForm'

const AuthForm = createAuthForm(ResetPasswordForm, 'Recover Your Password')

export const appendToken = (create, location) => values => {
  const urlParams = new URLSearchParams(location.search)

  return create({
    ...values,
    resetPasswordToken: urlParams.get('reset_password_token'),
  })
}

export default props => (
  <ResetPassword
    {...props}
    setCurrentUser={() => {}}
    render={renderProps => (
      <AuthForm
        {...renderProps}
        create={appendToken(renderProps.create, renderProps.location)}
      />
    )}
    renderCreated={() => <Redirect to="/" noThrow />}
  />
)
