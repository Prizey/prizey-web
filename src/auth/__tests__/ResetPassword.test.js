import React from 'react'
import renderer from 'react-test-renderer'

import ResetPasswordScreen, {
  appendToken,
} from '../ResetPasswordScreen/ResetPasswordScreen'

jest.mock('croods-auth', () => ({
  ResetPassword: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
      <div>renderCreated - {props.renderCreated({ user: { id: 1 } })}</div>
    </div>
  ),
}))

jest.mock('seasoned-auth-forms-web', () => ({
  NewPassword: props => (
    <div {...props}>NewPassword - {props.renderButton(props)}</div>
  ),
}))

jest.mock('@reach/router', () => ({
  Link: props => <div {...props}>Link - {props.children}</div>,
  Redirect: props => <div {...props}>Redirect - {props.children}</div>,
}))

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
      <div>renderCreated - {props.renderCreated({ id: 1 })}</div>
    </div>
  ),
}))

it('renders correctly', () => {
  const tree = renderer.create(<ResetPasswordScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('apppend a token on create', () => {
  const location = {
    search: 'reset_password_token=123123',
  }
  const data = {
    password: '123123',
  }
  const create = jest.fn()

  appendToken(create, location)(data)

  expect(create).toHaveBeenCalledWith({
    ...data,
    resetPasswordToken: '123123',
  })
})
