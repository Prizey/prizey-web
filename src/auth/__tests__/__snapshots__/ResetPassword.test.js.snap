// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  render={[Function]}
  renderCreated={[Function]}
  setCurrentUser={[Function]}
>
  New - 
  <div>
    render - 
    <div
      className="Component-root-9"
    >
      <div
        className="Component-header-7"
      />
      <div
        className="Component-container-4"
      >
        <img
          alt="Prizey"
          className="Component-logoImage-8"
          src="/logo.png"
        />
        <div
          className="Component-caption-2"
        />
        <div
          className="Component-children-3"
        >
          <h5
            className="MuiTypography-root-10 MuiTypography-h5-26 MuiTypography-alignCenter-33 Component-pageTitle-1"
          >
            Recover Your Password
          </h5>
          <div
            onSubmit={[Function]}
            renderButton={[Function]}
          >
            NewPassword - 
            <button
              className="MuiButtonBase-root-72 MuiButton-root-46 MuiButton-contained-57 MuiButton-containedPrimary-58 MuiButton-raised-60 MuiButton-raisedPrimary-61 MuiButton-fullWidth-71"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onSubmit={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              renderButton={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-47"
              >
                SET NEW PASSWORD
              </span>
              <span
                className="MuiTouchRipple-root-75"
              />
            </button>
          </div>
        </div>
        <div
          className="Component-footer-5"
        >
          <a
            className="Component-footerLink-6"
            href="https://app.termly.io/document/terms-and-conditions/4de5f3bb-a333-47bb-9daf-f82f7d4e1cfb"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-10 MuiTypography-caption-20 MuiTypography-colorTextSecondary-43 MuiTypography-alignCenter-33"
            >
              TERMS
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://app.termly.io/document/privacy-policy/8192f107-a2dc-4548-bf83-97907c6d83c3"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-10 MuiTypography-caption-20 MuiTypography-colorTextSecondary-43 MuiTypography-alignCenter-33"
            >
              PRIVACY
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-10 MuiTypography-caption-20 MuiTypography-colorTextSecondary-43 MuiTypography-alignCenter-33"
            >
              FAIRNESS
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="/all-prizes"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-10 MuiTypography-caption-20 MuiTypography-colorTextSecondary-43 MuiTypography-alignCenter-33"
            >
              ALL PRIZES
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    renderCreated - 
    <div
      noThrow={true}
      to="/"
    >
      Redirect - 
    </div>
  </div>
</div>
`;
